tests/ha/conftest.py

```diff
@@ -23,3 +29,87 @@ def setup_SmartSwitchHaTrafficTest(duthost, ptfhost, ptfadapter, vmhost, tbinfo)
     ha_io = SmartSwitchHaTrafficTest(activehost, standbyhost, ptfhost,
                                      ptfadapter, vmhost, tbinfo, io_ready, namespace="ns1")
     return ha_io
+
+
+def get_all_ptf_port_indices_from_mg_facts(mg_asic_facts):
+    """
+    Retrieve all PTF port indices from minigraph ASIC facts.
+
+    Args:
+        mg_asic_facts (list): List of (asic_index, asic_facts_dict) tuples.
+
+    Returns:
+        dict: {ptf_port_index: port_name}
+    """
+    all_port_indices = {}
+
+    for asic_index, asic_facts in mg_asic_facts:
+        ptf_indices = asic_facts.get('minigraph_ptf_indices', {})
+        for port_name, port_index in ptf_indices.items():
+            all_port_indices[port_index] = port_name
+
+    return all_port_indices
+
+
+def get_all_ptf_ports_from_all_duts(duts_minigraph_facts):
+    combined_ports = {}
+    for dut_hostname, mg_asic_facts_list in duts_minigraph_facts.items():
+        dut_ports = get_all_ptf_port_indices_from_mg_facts(mg_asic_facts_list)
+        for ptf_idx, port_name in dut_ports.items():
+            combined_ports[ptf_idx] = (dut_hostname, port_name)  # Include DUT
+    return combined_ports
+
+
+@pytest.fixture(scope="module")
+def setup_namespaces_with_routes(ptfhost, duthosts, duts_minigraph_facts):
+    ns_ifaces = []
+
+    # Get all PTF ports from all DUTs combined
+    all_ports = get_all_ptf_ports_from_all_duts(duts_minigraph_facts)
+    sorted_port_indices = sorted(all_ports.keys())
+
+    # Example split ports arbitrarily for namespace assignment
+    ns1_ports = sorted_port_indices[:2]
+    ns2_ports = sorted_port_indices[-2:]
+
+    for idx, port_idx in enumerate(ns1_ports, start=1):
+        iface_name = f"eth{port_idx}"
+        dut_name, _ = all_ports[port_idx]  # Unpack DUT name from all_ports
+        ns_ifaces.append({
+            "namespace": "ns1",
+            "iface": iface_name,
+            "ip": f"172.16.1.{idx}/24",
+            "next_hop": "172.16.1.254",
```

Comment:
Per our offline discussion, ns1 is used for sending (tcp_client), notice that your tcp_server is hosted as "172.16.1.1", please make sure ip is configured correctly here.